### PR TITLE
Improved client wrapper, CI, and a simple example

### PR DESCRIPTION
**Pull request by robo-corg**
This wraps all the grpc methods with a slightly nicer interface that does not require a `&mut Client` since cloning a grpc client is [cheap](https://docs.rs/tonic/latest/tonic/client/index.html#concurrent-usage). Also converts the auth interceptor from a closure into a struct so that it can be cloned. The end result is `Client` now implements clone, though users probably wont need it most of the time since it's methods are `&self`.

Aside: I think an even higher level api would be great but did not have the time and this might be something that can be done by another crate that takes this one as a dependency.

# Change
* Bump `tonic` to `0.7` and `prost` to `0.10` which are the last versions where protoc seems to magically get built.
* CI copied from https://github.com/actions-rs/example/blob/master/.github/workflows/quickstart.yml
* Added a simple based on the one from the readme example
* `Client` now implements `Clone`
* Manually wrapped the grpc methods to make things `&self`
* Updated readme example to work with api changes

### CODE REVIEW

File: build.rs [hunk 1]
```diff
@@ -21,7 +21,7 @@ fn main() -> Result<()> {
     // The toplevel types directory.
     let pb_dir: std::path::PathBuf = env::var("TRITON_PROTOBUF")
         .ok()
-        .unwrap_or(concat!(env!("CARGO_MANIFEST_DIR"), "/common/protobuf").to_string())
+        .unwrap_or_else(|| concat!(env!("CARGO_MANIFEST_DIR"), "/common/protobuf").to_string())
```
Comment (imalsogreg) [inline build.rs:24]: Good catch. These eager functions are so tricky.

**Review by imalsogreg — APPROVED** [review]
This looks awesome!

File: build.rs (hunk 1 above)
Comment (robo-corg) [reply build.rs:24]: Clippy caught it for me :paperclip:

**Review by vincentpun — APPROVED** [review]
Thank you thank you 😄

**Comment by imalsogreg** [comment]
@robo-corg Think it's ready to merge?

**Pull request merged**